src/helpers/message.ts

```diff
@@ -0,0 +1,125 @@
+import { IModify, IPersistence, IRead } from '@rocket.chat/apps-engine/definition/accessors';
+import { IRoom, RoomType } from '@rocket.chat/apps-engine/definition/rooms';
+import { IUser } from '@rocket.chat/apps-engine/definition/users';
+import { LayoutBlock } from '@rocket.chat/ui-kit';
+
+import { NotificationsController } from './notification';
+
+/**
+ * Gets a direct message room between bot and another user, creating if it doesn't exist
+ *
+ * @param read
+ * @param modify
+ * @param appUser
+ * @param username the username to create a direct with bot
+ * @returns the room or undefined if botUser or botUsername is not set
+ */
+export async function getDirect(
+  read: IRead,
+  modify: IModify,
+  appUser: IUser,
+  username: string
+): Promise<IRoom | undefined> {
+  const usernames = [appUser.username, username];
+  let room: IRoom;
+  try {
+    room = await read.getRoomReader().getDirectByUsernames(usernames);
+  } catch (error) {
+    this.app.getLogger().error(error);
+    return;
+  }
+
+  if (room) {
+    return room;
+  } else {
+    let roomId: string;
+
+    // Create direct room between appUser and username
+    const newRoom = modify
+      .getCreator()
+      .startRoom()
+      .setType(RoomType.DIRECT_MESSAGE)
+      .setCreator(appUser)
+      .setMembersToBeAddedByUsernames(usernames);
+    roomId = await modify.getCreator().finish(newRoom);
```

Comment:
also, I think this method can throw an exception, so you should probably try to make sure you return undefined if there was an error creating the room